TESTS/netsocket/README.md

```diff
@@ -1654,6 +1669,311 @@ through a dedicated socket
+
+**Test steps:**
+
+1.  Call `gethostbyname_async()` in a row 5 times with a different host names.
+2.  Verify that callback is called with correct parameters 5 times for first operation.
+3.  Repeat 100 times steps 1 to 2.
+
+**Expected result:**
+
+Callback is called with NSAPI_ERROR_OK and IP address 5 times for every repeat.
+
+### SYNCHRONOUS_DNS
+
+**Description:**
+
+Verify basic functionality of synchronous DNS. Call `NetworkInterface::gethostbyname()` with a valid host name and verify result.
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname()` with a valid host name.
+2.  Verify that address was resolved and return value was valid.
+
+**Expected result:**
+
+Return value is NSAPI_ERROR_OK and IP address is obtained from the function call.
+
+### SYNCHRONOUS_DNS_MULTIPLE
+
+**Description:**
+
+Verify basic functionality of synchronous DNS. Call `NetworkInterface::gethostbyname()` with a list of 6 host names and verify result.
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname()` with a list of 6 host names
+2.  Verify that each of the addresses was resolved and return value was valid.
+
+**Expected result:**
+
+Return value is NSAPI_ERROR_OK and IP addresses are obtained from the function call.
+
+### SYNCHRONOUS_DNS_CACHE
+
+**Description:**
+
+Verify that the caching of DNS results works correctly. Call `NetworkInterface::gethostbyname()` 5 times with the same host name and verify result after each request. For first request, cache shall not contain the host name. Verify that first request completes slower than the requests made after it (where the response is found from cache).
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname()` with a host name. For the first request, host name shall not be found from cache.
+2.  Verify that address was resolved and return value was valid.
+3.  Repeat the sequence 4 times using the same host name.
+4.  For each request, calculate how long time it takes for DNS query to complete.
+
+**Expected result:**
+
+Return value is NSAPI_ERROR_OK and IP address is obtained from the function call 5 times. First request shall complete slower than the requests made after it (where the response is found from cache).
+
+### SYNCHRONOUS_DNS_INVALID_HOST
+
+**Description:**
+
+Verify that DNS failure error is provided for invalid hosts. Call `NetworkInterface::gethostbyname()` in a row 6 times with a different host names. First, third and fifth host name shall be invalid.
+
+**Preconditions:**
+
+1.  Network interface is initialised.
+2.  Network connection is up.
+
+**Test steps:**
+
+1.  Call `gethostbyname()` in a row 6 times with a different host names. Host names shall not be found from cache. First, third and fifth host name shall be invalid.
```

Comment:
"...with different host names."

Also, I'm not sure what the rest of this line means.